WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/StoreSetupProgressView.swift

```diff
@@ -0,0 +1,90 @@
+
+struct StoreSetupProgressView: View {
+    let isExpanded: Bool
+
+    let totalNumberOfTasks: Int
+
+    let numberOfTasksCompleted: Int
+
+    var body: some View {
+        HStack(alignment: .top) {
+            VStack(alignment: isExpanded ? .center : .leading, spacing: Layout.verticalSpacing) {
+                // Title label
+                Text(Localization.title)
+                    .fontWeight(.semibold)
+                    .if(isExpanded) { $0.titleStyle() }
+                    .if(!isExpanded) { $0.headlineStyle() }
+                    .multilineTextAlignment(isExpanded ? .center : .leading)
+
+                // Progress view
+                ProgressView(value: Double(numberOfTasksCompleted), total: Double(totalNumberOfTasks))
+                    .tint(.init(uiColor: .accent))
+                    .frame(width: isExpanded ? Layout.ProgressView.widthExpanded : Layout.ProgressView.widthCollapsed, height: Layout.ProgressView.height)
+
+                // Subtitle label
+                Text(String(format: isExpanded ? Localization.TasksCompleted.expanded : Localization.TasksCompleted.collapsed,
+                            numberOfTasksCompleted,
+                            totalNumberOfTasks))
+                    .footnoteStyle()
+                    .multilineTextAlignment(isExpanded ? .center : .leading)
+            }
+
+            Spacer()
+                .renderedIf(!isExpanded)
+
+            // More button
+            Button {
+                // TODO: Show the popup with feedback button
+            } label: {
+                Image(uiImage: .ellipsisImage)
+                    .flipsForRightToLeftLayoutDirection(true)
+                    .foregroundColor(Color(.textTertiary))
+            }
+            .renderedIf(!isExpanded)
+        }
+    }
+}
+
+private extension StoreSetupProgressView {
+    enum Layout {
+        static let horizontalSpacing: CGFloat = 16
+        static let verticalSpacing: CGFloat = 8
+
+        enum ProgressView {
+            static let height: CGFloat = 6
+            static let widthCollapsed: CGFloat = 152.5
+            static let widthExpanded: CGFloat = 205.8
+        }
+    }
+
+    enum Localization {
+        static let title = NSLocalizedString(
+            "Store setup",
+            comment: "Title of the store setup progress view."
+        )
+
+        enum TasksCompleted {
+            static let collapsed = NSLocalizedString(
+                "%1$d/%2$d completed",
+                comment: "Shows how many tasks are completed in the store setup process." +
+                "%1$d represents the tasks completed. %2$d represents the total number of tasks."
+            )
+
+            static let expanded = NSLocalizedString(
+                "%1$d of %2$d tasks completed",
+                comment: "Shows how many tasks are completed in the store setup process." +
+                "%1$d represents the tasks completed. %2$d represents the total number of tasks."
```

Comment:
super nit: might be nice adding a note about the difference between these two strings for the translators.

Reply:
Done in https://github.com/woocommerce/woocommerce-ios/pull/8948/commits/8404088d17778e9c217ea5babd91ba4ba4c1ce0f